### PR TITLE
add bluepill stm32f103c8 support

### PR DESCRIPTION
**Pull request by zerog2k**
based on olimex-p103 bsp

### CODE REVIEW

File: hw/bsp/bluepill/bsp.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+bsp.compiler: '@apache-mynewt-core/compiler/arm-none-eabi-m3'
+
+bsp.linkerscript:
+    - 'hw/bsp/bluepill/bluepill.ld'
+    - '@apache-mynewt-core/hw/mcu/stm/stm32f1xx/stm32f103.ld'
+
+bsp.linkerscript.BOOT_LOADER.OVERWRITE:
+    - 'hw/bsp/bluepill/boot-bluepill.ld'
+    - '@apache-mynewt-core/hw/mcu/stm/stm32f1xx/stm32f103.ld'
+bsp.part2linkerscript: "hw/bsp/bluepill/split-bluepill.ld"
```
Comment (utzig) [inline hw/bsp/bluepill/bsp.yml]: Not sure why this is here, and `split-bluepill.ld` doesn't seem to be part of this PR. Missed including it?

File: hw/bsp/bluepill/bsp.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+bsp.downloadscript: "hw/bsp/bluepill/bluepill_download.sh"
+bsp.debugscript: "hw/bsp/bluepill/bluepill_debug.sh"
+bsp.downloadscript.WINDOWS.OVERWRITE: "hw/bsp/bluepill/bluepill_download.cmd"
+bsp.debugscript.WINDOWS.OVERWRITE: "hw/bsp/bluepill/bluepill_debug.cmd"
```
Comment (utzig) [inline hw/bsp/bluepill/bsp.yml]: The `*.cmd` files are not part of the PR as well (those can be copied verbatim from `olimex-p103`)

File: hw/bsp/bluepill/bsp.yml (hunk 1 above)
Comment (zerog2k) [reply hw/bsp/bluepill/bsp.yml]: not sure where I got this from, removed.

File: hw/bsp/bluepill/bsp.yml (hunk 2 above)
Comment (zerog2k) [reply hw/bsp/bluepill/bsp.yml]: thanks, missed that, copied over for our windows friends

**Comment by utzig** [comment]
There is no cookbook manual for configuring the `stm32YYxx_hal_conf.h` file but usually we disable all non-used modules. You could take a look at `hw/bsp/nucleo-f413zh/include/bsp/stm32f4xx_hal_conf.h` for an extreme example. The update based on config options sounds like a nice feature to have as well. This is not **that** important because it only slows down the build times slightly.

Also please rollback the `hw/bsp/bluepill/boot-bluepill.ld` license changes (don't mind the bot comments!).

**Comment by mkiiskila** [comment]
In theory you could make the hal_conf.h setting be dependent on the BSP config options. In practice, it's going to create a lot of messy code, and probably not that useful (i.e what Fabio said :).
In general, people would have their own BSP for the thing they're building, and they'd enable exactly the things they'd need for their board.

That being said, I can see why you would want to do the hal_conf.h dependent on the config options for this particular board. It is severely limited in size, and people would want to use the same BSP for different projects of theirs. So I think it would be reasonable to do that for this particular board. What do you think Fabio?

**Comment by utzig** [comment]
Just to be clear what I meant by "This is not that important because it only slows down the build times slightly." is that it is not that important to disable the non-used modules, programmatically or manually. It would not block a merge from me!

Answering @mkiiskila's question: I think it sounds cool to have that, but in practice I am not so sure because we would have one for each BSP, etc. Even if it's just this BSP, it still have some extra issues, like, now to be sure all options work, we would need to build each possible syscfg configuration that affects those module configurations. Also since we already garbage collects non-used symbols when linking, in practice this only makes a difference in the build times (because enabled modules still have to be built!). Not sure the effort of doing and maintaining this would bring much added value in return.

But I might as well be missing something or some different point of view! ;-)

**Comment by mkiiskila** [comment]
Yes; painful to maintain. I agree.

And you're right regarding link time garbage collection; if the init call to HAL from within BSP is commented out, the end result should not have any traces of those unused drivers.

**Comment by zerog2k** [comment]
yes, so I was also worried about whether it would link in unused code/symbols (sdcc seems to have a problem with this), but seems this does not:
```
jens@pluto:~/build/newt/myproj$ newt size f103_blinky
Size of Application Image: app
  FLASH     RAM 
     45     287 *fill*
    122       9 apps_blinky.a
    696      56 hw_bsp_bluepill.a
     52       0 hw_cmsis-core.a
    276       0 hw_drivers_uart_uart_hal.a
    148       0 hw_hal.a
   2096      64 hw_mcu_stm_stm32_common.a
   3130      32 hw_mcu_stm_stm32f1xx.a
   4644    8217 kernel_os.a
   1540      12 libc_baselibc.a
   2229     158 sys_console_full.a
    328     128 sys_flash_map.a
    412      12 sys_mfg.a
     32       5 sys_sysinit.a
     56       0 util_mem.a
     92       0 f103_blinky-sysinit-app.a
   1456       0 libgcc.a

objsize
   text	   data	    bss	    dec	    hex	filename
  17316	     60	   8952	  26328	   66d8	/home/jens/build/newt/myproj/bin/targets/f103_blinky/app/apps/blinky/blinky.elf
```

also, are you sure this really speeds up compile time?
So I just updated stm32f1xx_hal_conf.h in the local bsp of my project so it looks like this:
```
#define HAL_MODULE_ENABLED
#if 0
#define HAL_ADC_MODULE_ENABLED
#define HAL_CAN_MODULE_ENABLED
#define HAL_CEC_MODULE_ENABLED
#define HAL_CORTEX_MODULE_ENABLED
#define HAL_CRC_MODULE_ENABLED
#define HAL_DAC_MODULE_ENABLED
#define HAL_DMA_MODULE_ENABLED
#define HAL_ETH_MODULE_ENABLED
#define HAL_FLASH_MODULE_ENABLED
#define HAL_GPIO_MODULE_ENABLED
#define HAL_HCD_MODULE_ENABLED
#define HAL_I2C_MODULE_ENABLED
#define HAL_I2S_MODULE_ENABLED
#define HAL_IRDA_MODULE_ENABLED
#define HAL_IWDG_MODULE_ENABLED
#define HAL_NAND_MODULE_ENABLED
#define HAL_NOR_MODULE_ENABLED
#define HAL_PCCARD_MODULE_ENABLED
#define HAL_PCD_MODULE_ENABLED
#define HAL_PWR_MODULE_ENABLED
#define HAL_RCC_MODULE_ENABLED
#define HAL_RTC_MODULE_ENABLED
#define HAL_SD_MODULE_ENABLED
#define HAL_SMARTCARD_MODULE_ENABLED
#define HAL_SPI_MODULE_ENABLED
#define HAL_SRAM_MODULE_ENABLED
#define HAL_TIM_MODULE_ENABLED
#define HAL_UART_MODULE_ENABLED
#define HAL_USART_MODULE_ENABLED
#define HAL_WWDG_MODULE_ENABLED
#define HAL_MMC_MODULE_ENABLED
#else
#define HAL_USART_MODULE_ENABLED
#define HAL_UART_MODULE_ENABLED
#define HAL_TIM_MODULE_ENABLED
#define HAL_ADC_MODULE_ENABLED
#define HAL_DMA_MODULE_ENABLED
#define HAL_FLASH_MODULE_ENABLED
#define HAL_GPIO_MODULE_ENABLED
#define HAL_I2C_MODULE_ENABLED
#define HAL_IWDG_MODULE_ENABLED
#define HAL_RCC_MODULE_ENABLED
#define HAL_SPI_MODULE_ENABLED
#define HAL_CORTEX_MODULE_ENABLED
#endif
```
then build blinky again:
```
jens@pluto:~/build/newt/myproj$ newt build f103_blinky
Building target targets/f103_blinky
Compiling bin/targets/f103_blinky/generated/src/f103_blinky-sysflash.c
Compiling bin/targets/f103_blinky/generated/src/f103_blinky-sysinit-app.c
Compiling apps/blinky/src/main.c
Compiling hw/bsp/bluepill/src/hal_bsp.c
Assembling hw/bsp/bluepill/src/arch/cortex_m3/startup_stm32f103xb.s
Compiling hw/bsp/bluepill/src/system_stm32f1xx.c
Compiling repos/apache-mynewt-core/hw/drivers/uart/src/uart.c
Compiling repos/apache-mynewt-core/hw/cmsis-core/src/cmsis_nvic.c
Compiling repos/apache-mynewt-core/hw/hal/src/hal_common.c
Compiling repos/apache-mynewt-core/hw/hal/src/hal_flash.c
Compiling repos/apache-mynewt-core/hw/drivers/uart/uart_hal/src/uart_hal.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/hal_gpio.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/hal_i2c.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/hal_os_tick.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/hal_spi.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/hal_system_start.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/hal_system.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/hal_timer.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/hal_uart.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/hal_watchdog.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/sbrk.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/stm32_driver_mod_i2c_v1.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/stm32_driver_mod_i2c_v2.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/stm32_driver_mod_spi.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32_common/src/stm32_hw_id.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_adc_ex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_adc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_can.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_cec.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_cortex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_crc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_dac.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_dma.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_dac_ex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_eth.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_flash.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_flash_ex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_gpio.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_gpio_ex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_hcd.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_i2c.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_i2s.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_irda.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_iwdg.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_mmc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_nand.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_pccard.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_nor.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_pcd.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_pwr.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_pcd_ex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_rcc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_rcc_ex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_rtc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_rtc_ex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_sd.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_smartcard.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_spi.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_spi_ex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_sram.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_tim.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_tim_ex.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_uart.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_usart.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_wwdg.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_adc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_crc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_dac.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_dma.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_exti.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_gpio.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_fsmc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_i2c.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_pwr.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_rcc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_rtc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_spi.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_sdmmc.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_tim.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_usart.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_utils.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/ext/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_ll_usb.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/hal_flash.c
Compiling repos/apache-mynewt-core/kernel/os/src/endian.c
Compiling repos/apache-mynewt-core/hw/mcu/stm/stm32f1xx/src/hal_reset_cause.c
Compiling repos/apache-mynewt-core/kernel/os/src/os.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_callout.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_cputime.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_cputime_1mhz.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_cputime_high.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_cputime_pwr2.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_eventq.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_dev.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_heap.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_mbuf.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_mempool.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_msys_init.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_mutex.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_sched.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_sanity.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_sem.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_time.c
Compiling repos/apache-mynewt-core/kernel/os/src/os_task.c
Compiling repos/apache-mynewt-core/kernel/os/src/arch/cortex_m3/os_arch_arm.c
Compiling repos/apache-mynewt-core/kernel/os/src/arch/cortex_m3/os_fault.c
Assembling repos/apache-mynewt-core/kernel/os/src/arch/cortex_m3/m3/HAL_CM3.s
Assembling repos/apache-mynewt-core/kernel/os/src/arch/cortex_m3/m3/SVC_Table.s
Compiling repos/apache-mynewt-core/libc/baselibc/src/asprintf.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/atoi.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/atol.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/bsearch.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/atoll.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/bzero.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/calloc.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/fgets.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/inline.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/jrand48.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/lrand48.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/malloc.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memccpy.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memchr.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memcmp.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memcpy.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memfile.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memmem.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memmove.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memrchr.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memset.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/memswap.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/mrand48.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/nrand48.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/qsort.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/mynewt.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/sprintf.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/realloc.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/srand48.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/sscanf.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strcasecmp.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/start.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strcat.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strchr.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strcmp.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strcpy.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strcspn.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strdup.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strlcat.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strlcpy.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strlen.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strncasecmp.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strncat.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strncmp.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strncpy.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strndup.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strnlen.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strntoimax.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strpbrk.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strntoumax.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strrchr.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strsep.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strstr.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strspn.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strtok.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strtoimax.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strtok_r.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strtol.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strtoul.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strtoll.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strtoull.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/strtoumax.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/vasprintf.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/tinyprintf.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/vprintf.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/vsscanf.c
Compiling repos/apache-mynewt-core/libc/baselibc/src/vsprintf.c
Compiling repos/apache-mynewt-core/sys/console/full/src/ble_monitor_console.c
Compiling repos/apache-mynewt-core/sys/console/full/src/console.c
Compiling repos/apache-mynewt-core/sys/console/full/src/console_fmt.c
Compiling repos/apache-mynewt-core/sys/console/full/src/rtt_console.c
Compiling repos/apache-mynewt-core/sys/console/full/src/ticks.c
Compiling repos/apache-mynewt-core/sys/console/full/src/uart_console.c
Compiling repos/apache-mynewt-core/sys/flash_map/src/flash_map.c
Compiling repos/apache-mynewt-core/sys/mfg/src/mfg.c
Compiling repos/apache-mynewt-core/sys/sysinit/src/sysinit.c
Compiling repos/apache-mynewt-core/util/mem/src/mem.c
Archiving apps_blinky.a
Archiving f103_blinky-sysinit-app.a
Archiving hw_bsp_bluepill.a
Archiving hw_cmsis-core.a
Archiving hw_drivers_uart.a
Archiving hw_drivers_uart_uart_hal.a
Archiving hw_hal.a
Archiving hw_mcu_stm_stm32_common.a
Archiving hw_mcu_stm_stm32f1xx.a
Archiving kernel_os.a
Archiving libc_baselibc.a
Archiving sys_console_full.a
Archiving sys_flash_map.a
Archiving sys_mfg.a
Archiving sys_sysinit.a
Archiving util_mem.a
Linking /home/jens/build/newt/myproj/bin/targets/f103_blinky/app/apps/blinky/blinky.elf
Target successfully built: targets/f103_blinky

```

Looks like it's still compiling "disabled" hal modules?
(not really an stm32 hal/spl/cube guru)

**Comment by zerog2k** [comment]
here's a verbose gist of the clean and build:
https://gist.github.com/zerog2k/17695431154b6baba271e26313b50b89

looks like maybe several of these are getting pulled in from mcu includes?

**Comment by utzig** [comment]
@zero2k Yes, those files are still being compiled because they are seen by `newt` (the build tool) and the are passed to gcc. But if you open of those files and check their contents, they have some form of `#if HAL_XXX_MODULE_ENABLED` at the beginning which acts like a `#if 0` around the whole code inside the file.

**Comment by utzig** [comment]
> do you want me to resubmit as a single commit pr, or would you squash merge it?

Please do it!

**Comment by zerog2k** [comment]
resubmitted with single squashed commit in PR #1523
